include/nuttx/wireless/ieee80211/ieee80211.h

```diff
@@ -42,849 +42,4026 @@
+static inline uint8_t
+ieee80211_he_ppe_size(uint8_t ppe_thres_hdr, const uint8_t *phy_cap_info)
+{
+  uint8_t n;
 
-#define IEEE80211_AUTH_ALG_OPEN            0x0000
-#define IEEE80211_AUTH_ALG_SHARED          0x0001
-#define IEEE80211_AUTH_ALG_LEAP            0x0080
+  if ((phy_cap_info[6] &
+        IEEE80211_HE_PHY_CAP6_PPE_THRESHOLD_PRESENT) == 0)
+    return 0;
 
-/* WEP */
+  n = hweight8(ppe_thres_hdr &
+      IEEE80211_PPE_THRES_RU_INDEX_BITMASK_MASK);
+  n *= (1 + ((ppe_thres_hdr & IEEE80211_PPE_THRES_NSS_MASK) >>
+        IEEE80211_PPE_THRES_NSS_POS));
 
-#define IEEE80211_WEP_KEYLEN               5   /* 40bit */
-#define IEEE80211_WEP_NKID                 4   /* number of key ids */
-#define IEEE80211_CHALLENGE_LEN            128
+  /* Each pair is 6 bits, and we need to add the 7 "header" bits to the
+   * total size.
+   */
 
-/* WEP header constants */
+  n = (n * IEEE80211_PPE_THRES_INFO_PPET_SIZE * 2) + 7;
+  n = DIV_ROUND_UP(n, 8);
 
-#define IEEE80211_WEP_IVLEN                3   /* 24bit */
-#define IEEE80211_WEP_KIDLEN               1   /* 1 octet */
-#define IEEE80211_WEP_CRCLEN               4   /* CRC-32 */
-#define IEEE80211_CRC_LEN                  4
-#define IEEE80211_WEP_TOTLEN \
-    (IEEE80211_WEP_IVLEN + IEEE80211_WEP_KIDLEN + IEEE80211_WEP_CRCLEN)
+  return n;
+}
 
-/* 802.11i defines an extended IV for use with non-WEP ciphers.
- * When the EXTIV bit is set in the key id byte an additional
- * 4 bytes immediately follow the IV for TKIP.  For CCMP the
- * EXTIV bit is likewise set but the 8 bytes represent the
- * CCMP header rather than IV+extended-IV.
+/* HE Operation defines */
+
+#define IEEE80211_HE_OPERATION_DFLT_PE_DURATION_MASK  0x00000007
+#define IEEE80211_HE_OPERATION_TWT_REQUIRED           0x00000008
+#define IEEE80211_HE_OPERATION_RTS_THRESHOLD_MASK     0x00003ff0
+#define IEEE80211_HE_OPERATION_RTS_THRESHOLD_OFFSET   4
+#define IEEE80211_HE_OPERATION_VHT_OPER_INFO          0x00004000
+#define IEEE80211_HE_OPERATION_CO_HOSTED_BSS          0x00008000
+#define IEEE80211_HE_OPERATION_ER_SU_DISABLE          0x00010000
+#define IEEE80211_HE_OPERATION_6GHZ_OP_INFO           0x00020000
+#define IEEE80211_HE_OPERATION_BSS_COLOR_MASK         0x3f000000
+#define IEEE80211_HE_OPERATION_BSS_COLOR_OFFSET       24
+#define IEEE80211_HE_OPERATION_PARTIAL_BSS_COLOR      0x40000000
+#define IEEE80211_HE_OPERATION_BSS_COLOR_DISABLED     0x80000000
+
+/* ieee80211_he_6ghz_oper - HE 6 GHz operation Information field
+ * @primary: primary channel
+ * @control: control flags
+ * @ccfs0: channel center frequency segment 0
+ * @ccfs1: channel center frequency segment 1
+ * @minrate: minimum rate (in 1 Mbps units)
  */
 
-#define IEEE80211_WEP_EXTIV                0x20
-#define IEEE80211_WEP_EXTIVLEN             4    /* extended IV length */
-#define IEEE80211_WEP_MICLEN               8    /* trailing MIC */
+struct ieee80211_he_6ghz_oper
+{
+  uint8_t primary;
+#define IEEE80211_HE_6GHZ_OPER_CTRL_CHANWIDTH         0x3
```

Comment:
very strange place for definitions

Reply:
It is fine, this define just use for HE 6 GHz operation Information field

Reply:
I mean that it is very strange to place `#define` in the middle of the structure definition. If is not forbidden, but still strange

Reply:
Done